src/main/java/com/google/cloud/spanner/pgadapter/ConnectionHandler.java

```diff
@@ -80,7 +83,9 @@ public class ConnectionHandler extends Thread {
   private static final String CHANNEL_PROVIDER_PROPERTY = "CHANNEL_PROVIDER";
 
   private final ProxyServer server;
-  private final Socket socket;
+  private final ByteChannel socketChannel;
+  private final InetSocketAddress remoteAddress;
```

Comment:
Should we add a comment that this is nullable?

Reply:
Good point, added a `@Nullable` annotation and a comment.